### PR TITLE
add `gitops get profiles` command

### PR DESCRIPTION
**Pull request by aclevername**
Closes: #1108

**What changed?**
New command:
```
gitops --namespace test-namespace get profiles
NAME    DESCRIPTION     AVAILABLE_VERSIONS
podinfo Podinfo Helm chart for Kubernetes       6.0.0,6.0.1
```

To be able to see the available profiles

**How did you test it?**
Manually & acceptance test

**Release notes**
Add new `gitops get profiles` command for discovering available profiles

**Documentation Changes**
TODO

### CODE REVIEW

File: cmd/gitops/get/profiles/cmd.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	if err != nil {
+		return fmt.Errorf("error initializing kubernetes client: %w", err)
+	}
+
+	ns, err := cmd.Parent().Parent().Flags().GetString("namespace")
+	if err != nil {
+		return err
+	}
+
+	return profiles.GetProfiles(context.Background(), profiles.GetOptions{
```
[the remark this answers is not in the document]
Comment (nikimanoledaki) [reply cmd/gitops/get/profiles/cmd.go:55]: basically I'm wondering whether it's better to list the profiles that are in the cluster through the local kubeconfig or through the `HelmRelease`s in `/.weave-gitops/clusters/<cluster name>/system`, or both?

[the remark this answers is not in the document]
Comment (aclevername) [reply cmd/gitops/get/profiles/cmd.go:55]: this is listing *available* profiles, as in what you can install (the charts that are availble from the `HelmRepository` resource), not what you've actually got installed.

[the remark this answers is not in the document]
Comment (bigkevmcd) [reply cmd/gitops/get/profiles/cmd.go:55]: I guess there are three possible lists of Profiles...

 1. List of Profiles in a cluster directory in a Git repository
 2. List of available profiles (to be installed)
 3. List of installed profiles in the cluster pointed at by your current context

So, we need to figure out how to be able to query for these

**Review by Skarlso — APPROVED** [review]
Does what it should 👍

File: pkg/services/profiles/get.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal response: %w", err)
+	}
+
+	printProfiles(profiles, opts.Writer)
+
+	return nil
+}
+
+func printProfiles(profiles *pb.GetProfilesResponse, w io.Writer) {
```
[the remark this answers is not in the document]
Comment (aclevername) [reply pkg/services/profiles/get.go:47]: yarp

File: cmd/gitops/get/profiles/cmd.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aclevername) [reply cmd/gitops/get/profiles/cmd.go:55]: This story/PR is definitely just for `2.`, but I agree we will need `1.` and `3.` at some point

[the remark this answers is not in the document]
Comment (nikimanoledaki) [reply cmd/gitops/get/profiles/cmd.go:55]: Sounds good! @Himangini and I will link up on Monday to create some follow-up tickets.

[the remark this answers is not in the document]
Comment (Himangini) [reply cmd/gitops/get/profiles/cmd.go:55]: > I guess there are three possible lists of Profiles...
> 
> 1. List of Profiles in a cluster directory in a Git repository
> 2. List of available profiles (to be installed)
> 3. List of installed profiles in the cluster pointed at by your current context
> 
> So, we need to figure out how to be able to query for these

This should've been discussed in planning.

[the remark this answers is not in the document]
Comment (Himangini) [reply cmd/gitops/get/profiles/cmd.go:55]: > This story/PR is definitely just for `2.`, but I agree we will need `1.` and `3.` at some point

@aclevername can you pair with @bigkevmcd and come up with tickets for `1.` and `3.` with details of what's required, please. Please also link this PR for reference in the new ones.

**Review by nikimanoledaki — APPROVED** [review]
LGTM!

File: cmd/gitops/get/profiles/cmd.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (aclevername) [reply cmd/gitops/get/profiles/cmd.go:55]: https://github.com/weaveworks/weave-gitops/issues/1318

**Pull request merged**